src/Popova/Task5/main.cpp

```diff
@@ -0,0 +1,179 @@
+#include <QVector3D>
+#include <cmath>
+#include "Types.hpp"
+#include "Sphere.hpp"
+#include "Plane.hpp"
+#include <QImage>
+#include <QColor>
+#include <QOpenGLFunctions>
+#include "QCoreApplication"
+#include <QCommandLineParser>
+#include <iostream>
+
+int WIDTH = 640;
+int HEIGHT = 480;
+constexpr float PI = 3.141592F;
+constexpr auto MAX_REFLECTIONS = 4;
+constexpr QVector3D BACKGROUND_COLOR{0.7f, 0.7f, 0.9f};
+constexpr QVector3D LIGHT_COLOR{1.0F, 1.0F, 1.0F};
+constexpr QVector3D LIGHT_POS{0.0F, 4.0F, 0.0F};
+constexpr float LIGHT_INTENSITY = 1.5F;
+constexpr QVector3D CAMERA_POS{0.0F, 1.0F, 3.0F};
+
+QVector3D mix(const QVector3D &a, const QVector3D &b, const float k){
+    return a * k + b*(1 - k);
+}
+
+
+QVector3D reflect(const QVector3D &incident, const QVector3D &normal) {
+return incident - 2.f * normal * (QVector3D::dotProduct(incident, normal));
+}
+
+
+QVector3D refract(const QVector3D &incident, const QVector3D &normal, const float n_1, const float n_2=1.f) {
+    auto cos_1 = - std::max(-1.f, std::min(1.f, QVector3D::dotProduct(incident, normal)));
```

Comment:
```suggestion
    const auto cos_1 = - std::max(-1.f, std::min(1.f, QVector3D::dotProduct(incident, normal)));
```
и во всем остальном коде

Reply:
fixed